### ISSUE TITLE
[Bug] Inference tab renders normalized tensors without inverse normalization.

### ISSUE THREAD

**Issue by NANDA2726** [JdeRobot/dl-DetectionSuite#524]
The Inference tab in `tabs/inference.py` displays the normalized image tensor directly without applying inverse normalization before rendering:
```python
img_to_draw = transforms.ToPILImage()(sample_tensor[0])
```

This causes the output image to appear with distorted, psychedelic colors instead of the original image, making the inference visualization incorrect and misleading. The image tensor should be de-normalized before display so the output image appears correctly with original colours.

## Proposed Fix
Add a `denormalize` helper function in `tabs/inference.py` and apply it before rendering:
```python
def denormalize(tensor, mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]):
    mean = torch.tensor(mean).view(3, 1, 1)
    std = torch.tensor(std).view(3, 1, 1)
    return torch.clamp(tensor.cpu() * std + mean, 0, 1)

denormalized = denormalize(sample_tensor[0])
img_to_draw = transforms.ToPILImage()(denormalized)
```

**Comment by dpascualhe** [comment]
Can you add a screenshot showing the reported behavior?

**Comment by NANDA2726** [comment]
Here is the screenshot:
<img width="1227" height="785" alt="Image" src="https://github.com/user-attachments/assets/ffbdaf21-26fb-4ef4-84f1-1f325c854c6a" />
The image appears to be normalized without inverse normalization, resulting in distorted colors during inference visualization.

**Comment by dpascualhe** [comment]
Hi Nanda, can you test it again with the lastest version? No other user has been suffering this problem. If you cannot replicate anymore, we will close the issue

**Comment by divyanshAgarwal123** [comment]
Hi, I checked this against the current master branch and the problem is still present in the refactored image-detection inference path. The UI renders the normalized sample tensor, even though predict() already maps the detection boxes back to the original image coordinates.

I reproduced the color change with a small RGB sample. I would like to fix this by drawing detections on the original uploaded RGB image and add a focused regression test so normalized tensors are not used for display. Please let me know if this scope is available for me to work on.

**Comment by dpascualhe** [comment]
Hi @divyanshAgarwal123 , it is working for me and other contributors. Could you send a small video with a screen capture of the issue you describe?

**Comment by divyanshAgarwal123** [comment]
Hi @dpascualhe — thanks for asking. Here is a 14-second capture of the full Streamlit app, with the complete result image visible:

https://github.com/user-attachments/assets/67f7af92-f86e-46bd-944c-d9785b777a9e

I used the same uploaded image and small fixed TorchScript detector in all three runs. The detector goes through the app's model loader and `TorchImageDetectionModel.predict()`; it is a controlled test fixture, not NANDA's original model.

1. Current `master`, config **without normalization**: colors look correct.
2. Current `master`, config **with ImageNet normalization** (mean 0.485/0.456/0.406; std 0.229/0.224/0.225): the displayed image becomes psychedelic.
3. PR #631, **same normalized config**: the original RGB colors return and the detection box remains.

The difference is conditional: manual configuration does not add normalization, which may explain why it works in other setups. On `master`, the inference tab converts the normalized `sample_tensor[0]` directly to a display image. #631 instead draws detections on the uploaded RGB image; `predict()` already maps boxes back to its coordinates. The focused GUI/image tests pass locally (12/12).
